packages/nx/src/utils/create-package-json.ts

```diff
@@ -106,62 +176,64 @@ export function createPackageJson(
 function findAllNpmDeps(
   projectName: string,
   graph: ProjectGraph,
-  list: {
-    dependencies: Record<string, string>;
-    peerDependencies: Record<string, string>;
-    peerDependenciesMeta: Record<string, { optional: boolean }>;
-  } = { dependencies: {}, peerDependencies: {}, peerDependenciesMeta: {} },
-  seen = new Set<string>()
-) {
-  const node = graph.externalNodes[projectName];
-
-  if (seen.has(projectName)) {
-    // if it's in peerDependencies, move it to regular dependencies
-    // since this is a direct dependency of the project
-    if (node && list.peerDependencies[node.data.packageName]) {
-      list.dependencies[node.data.packageName] = node.data.version;
-      delete list.peerDependencies[node.data.packageName];
-    }
-    return list;
-  }
+  list: NpmDeps,
+  seen: Set<string>,
+  dependencyPatterns: string[],
+  rootPatterns?: string[]
+): void {
+  const projectNode = graph.nodes[projectName];
+
+  const projectFiles = filterUsingGlobPatterns(
+    projectNode.data.root,
+    projectNode.data.files,
+    rootPatterns ?? dependencyPatterns
+  );
 
-  seen.add(projectName);
+  const projectDependencies = new Set<string>();
 
-  if (node) {
-    list.dependencies[node.data.packageName] = node.data.version;
-    recursivelyCollectPeerDependencies(node.name, graph, list, seen);
-  } else {
-    // we are not interested in the dependencies of external projects
-    graph.dependencies[projectName]?.forEach((dep) => {
-      if (dep.type === 'static' || dep.type === 'dynamic') {
-        findAllNpmDeps(dep.target, graph, list, seen);
+  projectFiles.forEach((fileData) =>
+    fileData.deps?.forEach((dep) => projectDependencies.add(dep))
```

Comment:
Using `deps` is depreacated and replaced with `dependencies`:

https://github.com/nrwl/nx/blob/master/packages/nx/src/config/project-graph.ts#L16

Please replace this line with:
```
fileData.dependencies?.forEach((dep) => projectDependencies.add(dep.target))
```